match/lib/match/git_helper.rb

```diff
@@ -80,21 +81,45 @@ def self.generate_commit_message(params)
     end
 
     def self.match_version(workspace)
-      path = File.join(workspace, "match_version.txt")
+      path = File.join(workspace, MATCH_VERSION_FILE_NAME)
       if File.exist?(path)
         Gem::Version.new(File.read(path))
       end
     end
 
-    def self.commit_changes(path, message, git_url, branch = "master")
+    def self.commit_changes(path, message, git_url, branch = "master", files_to_commmit = nil)
+      files_to_commmit ||= []
       Dir.chdir(path) do
         return if `git status`.include?("nothing to commit")
 
         Encrypt.new.encrypt_repo(path: path, git_url: git_url)
-        File.write("match_version.txt", Fastlane::VERSION) # unencrypted
-
         commands = []
-        commands << "git add -A"
+
+        if files_to_commmit.count > 0 # e.g. for nuke this is treated differently
+          if !File.exist?(MATCH_VERSION_FILE_NAME) || File.read(MATCH_VERSION_FILE_NAME) != Fastlane::VERSION.to_s
+            files_to_commmit << MATCH_VERSION_FILE_NAME
+            File.write(MATCH_VERSION_FILE_NAME, Fastlane::VERSION) # stored unencrypted
```

Comment:
Is all of this code in this `if` block to just keep track of what version of _match_ was used to encrypt the stuff?

Reply:
Yep 👍